zrml/prediction-markets/src/lib.rs

```diff
@@ -1687,6 +1692,83 @@ mod pallet {
 
             Ok(Some(weight).into())
         }
+
+        /// Allows the manual opening and closing for "broken" markets.
+        /// A market is "broken", if an unexpected chain stall happened
+        /// and the auto open / close was scheduled during this time.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(max{n, m})`,
+        /// where `n` is the number of market ids,
+        /// which open at the same time as the specified market,
+        /// and `m` is the number of market ids,
+        /// which close at the same time as the specified market.
+        #[pallet::call_index(21)]
+        #[pallet::weight((
+            T::WeightInfo::manually_open_market(CacheSize::get())
+                .max(T::WeightInfo::manually_close_market(CacheSize::get())),
+            Pays::Yes
+        ))]
+        #[transactional]
+        pub fn manually_open_or_close_market(
```

Comment:
Why don't we split this into two functions. You're basically writing two functions into one here.

Reply:
We can, it's just the style how you want to use it. If you split it, you need to know which markets are not yet closed and what market are not yet opened. If you have the combined function, you literally don't care about this distinction and just fix a "broken" market.